presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -349,14 +356,22 @@ public PrestoSparkTaskExecutorFactory(
 
         log.info(PlanPrinter.textPlanFragment(fragment, functionAndTypeManager, session, true));
 
-        MemoryPool memoryPool = new MemoryPool(new MemoryPoolId("spark-executor-memory-pool"), getQueryMaxTotalMemoryPerNode(session));
+        DataSize maxUserMemory = new DataSize(Math.min(nodeMemoryConfig.getMaxQueryMemoryPerNode().toBytes(), getQueryMaxMemoryPerNode(session).toBytes()), BYTE);
+        DataSize maxTotalMemory = new DataSize(Math.min(nodeMemoryConfig.getMaxQueryTotalMemoryPerNode().toBytes(), getQueryMaxTotalMemoryPerNode(session).toBytes()), BYTE);
+        DataSize maxBroadcastMemory = new DataSize(
```

Comment:
I recommend keeping this property as a hard override, as it is easier to comprehend it's semantics then. If the property is set - let's completely override the others, e.g.:

```
DataSize maxBroadcastMemory = getSparkBroadcastJoinMaxMemoryOverride(session);
if(maxBroadcastMemory == null) {
  maxBroadcastMemory = new DataSize(min(nodeMemoryConfig.getMaxQueryBroadcastMemory().toBytes(), getQueryMaxBroadcastMemory(session).toBytes()))
}
```

(don't forget to update the  `PrestoSparkQueryExecutionFactory`)